src/server/generic_family.cc

```diff
@@ -625,7 +625,16 @@ void OpScan(const OpArgs& op_args, const ScanOpts& scan_opts, uint64_t* cursor,
   string scratch;
   do {
     cur = prime_table->Traverse(
-        cur, [&](PrimeIterator it) { cnt += ScanCb(op_args, it, scan_opts, &scratch, vec); });
+        cur, [&](PrimeIterator it) { cnt += ScanCb(op_args, it, scan_opts, &scratch, vec); },
+        [](size_t count) -> size_t {
+          // Our limit is 100k keys. Estimate how many buckets that is.
+          const size_t limit = 100000 / (PrimeTable::kBucketNum * PrimeTable::kSlotNum);
```

Comment:
Rough estimate. Does not include stash buckets. Furthermore, we traverse logical buckets which span over the bid, adjacent buckets and stash buckets.